includes/model/class-post.php

```diff
@@ -305,41 +305,16 @@ public function get_attachments() {
 
 		// get URLs for each image
 		foreach ( $image_ids as $id ) {
-			$alt = \get_post_meta( $id, '_wp_attachment_image_alt', true );
-
-			/**
-			 * If you use the Jetpack plugin and its Image CDN, aka Photon,
-			 * the image strings returned will use the Photon URL.
-			 * We don't want that since Fediverse instances already do caching on their end.
-			 * Let the CDN only be used for visitors of the site.
-			 *
-			 * Old versions of Jetpack used the Jetpack_Photon class to do this.
-			 * New versions use the Image_CDN class.
-			 * Let's handle both.
-			 */
-			if ( \class_exists( '\Automattic\Jetpack\Image_CDN\Image_CDN' ) ) {
-				\remove_filter( 'image_downsize', array( \Automattic\Jetpack\Image_CDN\Image_CDN::instance(), 'filter_image_downsize' ) );
-			} elseif ( \class_exists( 'Jetpack_Photon' ) ) {
-				\remove_filter( 'image_downsize', array( \Jetpack_Photon::instance(), 'filter_image_downsize' ) );
-			}
-
-			$thumbnail = \wp_get_attachment_image_src( $id, 'full' );
-
-			// Re-enable Photon now that the image URL has been built.
-			if ( \class_exists( '\Automattic\Jetpack\Image_CDN\Image_CDN' ) ) {
-				\add_filter( 'image_downsize', array( \Automattic\Jetpack\Image_CDN\Image_CDN::instance(), 'filter_image_downsize' ), 10, 3 );
-			} elseif ( \class_exists( 'Jetpack_Photon' ) ) {
-				\add_filter( 'image_downsize', array( \Jetpack_Photon::instance(), 'filter_image_downsize' ), 10, 3 );
-			}
-
-			$mimetype = \get_post_mime_type( $id );
-
+			$thumbnail = $this->get_image( $id );
 			if ( $thumbnail ) {
-				$image = array(
-					'type' => 'Image',
-					'url' => $thumbnail[0],
+				$mimetype = \get_post_mime_type( $id );
+				$alt      = \get_post_meta( $id, '_wp_attachment_image_alt', true );
```

Comment:
Noting that this change moves those 2 functions inside the `$thumbnail` check, so they only run when necessary. Before this PR they ran regardless of the response from `wp_get_attachment_image_src`.